### ISSUE TITLE
Jest unit test and Electron

### ISSUE THREAD

**Issue by FrenchBen** [atom/electron#3909]
I've been using Jest for unit testing and currently running into the following error after updating my code to the latest electron-prebuilt of 0.36.0 on mac OSX:
Error from `npm test`
`node_modules/electron-prebuilt/dist/Electron.app/Contents/Resources/default_app/main.js: Cannot read property 'on' of undefined`

Current .js code:
```js
import electron from 'electron';
const remote = electron.remote;
const dialog = remote.dialog;
const app = remote.app;
```

previous code which passed the test:
```js
import remote from 'remote';
var dialog = remote.require('dialog');
var app = remote.require('app');
```

It seems that electron complains about the `app` object which is undefined during the test?

**Comment by FrenchBen** [comment]
added a mock to make it pass

**Comment by FrenchBen** [comment]
@ahanusa I believe I added this to fix it 
`__mocks__/electron.js`
```
module.exports = {
  require: jest.genMockFunction(),
  match: jest.genMockFunction(),
  app: jest.genMockFunction(),
  remote: jest.genMockFunction(),
  dialog: jest.genMockFunction()
};
```

**Comment by ahanusa** [comment]
awesome, worked like a champ.  Thanks a ton!

**Comment by FrenchBen** [comment]
:+1:

**Comment by mlynch** [comment]
@FrenchBen works great, thanks. For future reference: I had to add my `__mocks__` folder to `testPathIgnorePatterns` otherwise jest thought it was a test suite file.

**Comment by FrenchBen** [comment]
amazing that 2 years later this hack is still needed 😓

**Comment by centerorbit** [comment]
Not sure if this is the "right way", but `jest.genMockFunction(),` has been replaced with `jest.fn(),` (see: https://github.com/facebook/jest/issues/1018). I was using `getPath()`, so I had to nest the mock pretty deep, it didn't seem to cover all the way down unless I did it this way.

I did not need to apply @mlynch's ignore pattern for mine to work.

Here's my setup that worked:
```
module.exports = {
  require: jest.fn(),
  match: jest.fn(),
  app: jest.fn(),
  remote: {
    app: {
      getPath: jest.fn(),
    },
  },
  dialog: jest.fn(),
};

```
